src/layer/filter.rs

```diff
@@ -2,17 +2,15 @@ use tracing::Subscriber;
 use tracing_subscriber::{layer::Filter, registry::LookupSpan};
 
 use crate::{
-    native::{EventWriter, ProviderTypes},
+    native::{OutputMode, ProviderTraits},
     statics::get_event_metadata,
 };
 
-use super::*;
+use super::EtwFilter;
 
-impl<S, Mode> Filter<S> for EtwFilter<S, Mode>
+impl<S, OutMode: OutputMode> Filter<S> for EtwFilter<S, OutMode>
 where
     S: Subscriber + for<'a> LookupSpan<'a>,
-    Mode: ProviderTypes + 'static,
-    Mode::Provider: EventWriter<Mode> + 'static,
```

Comment:
These 2 constraints in the old code were the main source of confusion and difficulty following the internal logic.

Reply:
`Mode: ProviderTypes` - the Mode generic type implements the ProviderTypes trait. `Mode::Provider: EventWriter<Mode>` - the `Provider` associated type of the ProviderTypes trait implements EventWriter<Mode> trait. EventWriter will then have the same two constraints on its generics.